config/secure-headers.php

```diff
@@ -380,38 +560,17 @@
 				'blob:',
```

Comment:
Same thing here. If we don't use `data:` and `blob:` we should remove it.

Reply:
Removed in [3f5583f](https://github.com/LycheeOrg/Lychee/pull/1528/commits/3f5583f3f4300551dbb90229ed1d7e908202272d)

Reply:
I think we need `data:`

Reply:
I can't find the source, but there's a gif defined that way in `main.js`

Reply:
Unresolving this until someone can confirm where that `data:` gif is from and that we definitely don't use it.

Reply:
I haven't found any `data:` URL in `/resources` nor `Lychee-front/scripts`. So it is definitely not used by us directly. The `data:` scheme was added by [commit de5da01c](https://github.com/LycheeOrg/Lychee/pull/339) and it commented with

> Allow image to be directly encoded at the img source parameter

See

https://github.com/LycheeOrg/Lychee/blob/de5da01ca5a8e39ef82630ec9e174fb26b12d684/config/secure-headers.php#L380-L386

So it might be internally used by the 3rd-party library which renders the map. The only frontend code which was introduced by the corresponding frontend PR https://github.com/LycheeOrg/Lychee-front/pull/144 and might be suspect to using `data:` URLs is

https://github.com/LycheeOrg/Lychee-front/blob/3b71fe061dceaeaa507a0ad252d7f7351a7e15aa/scripts/main/view.js#L735-L754

This snippet of code renders the tiny map in the sidebar of a geo-tagged photo and adds the direction cone as an map overlay. I guess we can just test whether this part still works if `data:` URI are disabled. If it does, we should be fine.

Reply:
Actually it looks like it's only a 1x1px gif. It looks like openstreetmap might use this, and some libraries/etc use it to assist with lazy loading of images.

Based on that I don't mind removing it, so long as we remember that weird behaviour may be because of that. IIRC it should show up in the console when items are blocked by CSP, which may help to confirm.

Reply:
Oh, just saw your commit. I'll resolve this :P